components/script/dom/htmlmediaelement.rs

```diff
@@ -1220,6 +1234,118 @@ impl HTMLMediaElement {
+                        if audio_track.enabled() || self.AudioTracks().enabled_index().is_none() {
+                            let window = window_from_node(self);
+                            let task_source = window.task_manager().media_element_task_source();
+                            let this = Trusted::new(self);
+
+                            task_source
+                                .queue(
+                                    task!(media_track_change: move || {
+                                        let this = this.root();
+                                        this.AudioTracks().set_enabled(0, true);
+                                        this.upcast::<EventTarget>().fire_event(atom!("change"));
+                                    }),
+                                    window.upcast(),
+                                )
+                                .unwrap();
+                        }
+
+                        // Steps 7.
+                        TrackEvent::new(
+                            &self.global(),
+                            atom!("addtrack"),
+                            false,
+                            false,
+                            &Some(VideoTrackOrAudioTrackOrTextTrack::AudioTrack(audio_track)),
+                        );
+                    }
+                }
+
+                // => If the media resource is found to have a video track
+                if !metadata.video_tracks.is_empty() {
+                    for (i, _track) in metadata.video_tracks.iter().enumerate() {
+                        // Step 1.
+                        let kind = match i {
+                            0 => "main",
+                            _ => "",
+                        };
+                        let window = window_from_node(self);
+                        let video_track = VideoTrack::new(
+                            &window,
+                            DOMString::new(),
+                            DOMString::from(kind),
+                            DOMString::new(),
+                            DOMString::new(),
+                        );
+
+                        // Step 2.
+                        self.VideoTracks().add(&video_track);
+
+                        // Step 3.
+                        video_track.set_selected(false);
+
+                        // Step 4.
+                        // https://www.w3.org/TR/media-frags/#media-fragment-syntax
+                        // https://github.com/servo/servo/issues/22366
+
+                        // Step 5. & 6.
+                        if video_track.selected() || self.VideoTracks().selected_index().is_none() {
+                            let window = window_from_node(self);
+                            let task_source = window.task_manager().media_element_task_source();
+                            let this = Trusted::new(self);
+
+                            task_source
+                                .queue(
```

Comment:
Same comments about this code as the audio one per https://html.spec.whatwg.org/multipage/media.html#dom-videotrack-selected. The event code should be part of set_selected, rather than this algorithm.